src/relations/mysql_provider.py

```diff
@@ -256,3 +264,34 @@ def _on_database_broken(self, event: RelationBrokenEvent) -> None:
         except (MySQLDeleteUsersForRelationError, KeyError):
             logger.error(f"Failed to delete user for relation {relation_id}")
             return
+
+    def _on_database_provides_relation_departed(self, event: RelationDepartedEvent) -> None:
+        """Remove MySQL Router cluster metadata & router user for departing unit."""
+        if not self.charm.unit.is_leader():
+            return
+        if event.departing_unit.app.name == self.charm.app.name:
+            return
+
+        users = self.charm._mysql.get_mysql_router_users_for_unit(
+            relation_id=event.relation.id, mysql_router_unit_name=event.departing_unit.name
+        )
+        if not users:
+            return
+
+        if len(users) > 1:
```

Comment:
couldnt there be more than 1 applications related to the router? if so, wouldnt there be more than 1 user returned?

Reply:
> The unit name contains the app name

if that's the case, should we rename `created_by_juju_unit` to `created_by_router_identifier` where `router_identifier` is a combination of the juju model, application and unit for the router?